### ISSUE TITLE
GitHub Actions: Use new pull_request_target event

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#765]
GitHub Actions has introduced a new event type: `pull_request_target`

> https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target
> This event is similar to `pull_request`, except that it runs in the context of the base repository of the pull request, rather than in the merge commit. This means that you can more safely make your secrets available to the workflows triggered by the pull request, because only workflows defined in the commit on the base repository are run. For example, this event allows you to create workflows that label and comment on pull requests, based on the contents of the event payload.

> In order to solve this [security issues with `pull_request`], we’ve added a new `pull_request_target` event, which behaves in an almost identical way to the pull_request event with the same set of filters and payload. However, instead of running against the workflow and code from the merge commit, the event runs against the workflow and code from the base of the pull request. This means the workflow is running from a trusted source and is given access to a read/write token as well as secrets enabling the maintainer to safely comment on or label a pull request.

Switching this in to the `.github/workflows/branch.yml` workflow should mean that we can have automated comments for PRs coming from forks, rather than just branches 🎉 

I would like to also use it in `linting.yml`, which would that we could have the nice automated comment with the `nf-core lint` results from fork PRs. However, I think that we want lint tests to run on the merge commit code, so we probably shouldn't switch this one 😞

**Comment by ewels** [comment]
NB: Tricky to test, as needs to be merged to `master` in order to be active.

**Comment by ewels** [comment]
This blog post describes how we could get the lint comment to work: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

> The new `workflow_run` event enables you to trigger a new workflow when one or more workflows are requested or completed. Runs triggered by the `workflow_run` event always use the default branch for the repository, and have access to a read/write token as well as secrets.
> ***As an example, as a maintainer you could set up a workflow that takes the artifacts generated by the pull request workflow, do some analysis, and post comments back to the pull request.***

There must be a blog post or example somewhere that we can pinch to try this out..

**Comment by ewels** [comment]
Tested the linting with the new event trigger and it didn't work, as expected: https://github.com/nf-core/testpipeline/pull/7

So now we know..

**Comment by ewels** [comment]
I started looking into the `workflow_run` thing but it's actually fairly complex. The code I wrote to post the comments is not trivial:

https://github.com/nf-core/tools/blob/9b3eec9973160cd8b01e182b8f77ae7de2a73ddc/nf_core/lint.py#L1473-L1525

Specifically, it fetches the PR comments to see if there is a linting comment already. If there is, it updates it. If there isn't, it posts a new one. We'd need to figure out how to do this using a GitHub Actions workflow separate from the `nf-core lint` command if we take this new approach.

Editing instead of reposting is important as otherwise there will be a comment for every commit which is super spammy.
